src/utils/plotSpecs/generateFeaturesVsUMIsScatterplot.js

```diff
@@ -1,124 +1,135 @@
-const generateSpec = (config, plotData) => ({
-  $schema: 'https://vega.github.io/schema/vega/v5.json',
-  width: config.dimensions.width,
-  height: config.dimensions.height,
-  autosize: { type: 'fit', resize: true },
-  padding: 5,
+import calculateStdev from '../mathFormulas';
 
-  data: [
-    {
-      name: 'plotData',
-      values: plotData,
-      transform: [
-        {
-          type: 'filter',
-          expr: "datum['log_genes'] != null && datum['log_molecules'] != null",
-        },
-      ],
-    },
-  ],
+const generateSpec = (config, plotData) => {
+  const stdev = calculateStdev(plotData.map((p) => p.log_genes));
+  const lowerCutoff = Math.min(...plotData.map((p) => p.lower_cutoff)) - stdev;
+  const upperCutoff = Math.max(...plotData.map((p) => p.upper_cutoff)) + stdev;
+
+  return {
+    $schema: 'https://vega.github.io/schema/vega/v5.json',
+    width: config.dimensions.width,
+    height: config.dimensions.height,
+    autosize: { type: 'fit', resize: true },
+    padding: 5,
+
+    data: [
+      {
+        name: 'plotData',
+        values: plotData,
+        transform: [
+          {
+            type: 'filter',
+            expr: "datum['log_genes'] != null && datum['log_molecules'] != null",
+          },
+          {
+            type: 'filter',
+            expr: `datum.log_genes >= ${lowerCutoff} && datum.log_genes <= ${upperCutoff}`,
```

Comment:
since this expression is more involved, what about moving to a variable `logGenesWithinCutoff` or something? it makes the expressions more readable, kind of in the same way that `generateMitochondrialFractionHistogram`'s  expression work.

Reply:
I think this one is pretty clear as it's a pretty common expression in maths (e.g. `b > a and b < c` )

Reply:
I think this one is pretty clear as it's a pretty common expression in maths (e.g. b > a and b < c )